component/otelcol/config_filter.go

```diff
@@ -0,0 +1,215 @@
+}
+
+// MatchProperties specifies the set of properties in a spans/log/metric to match
+// against and if the input data should be included or excluded from the processor.
+type MatchProperties struct {
+	MatchType    string        `river:"match_type,attr"`
+	RegexpConfig *RegexpConfig `river:"regexp,block,optional"`
+
+	// Note: For spans, one of Services, SpanNames, Attributes, Resources or Libraries must be specified with a
+	// non-empty value for a valid configuration.
+
+	// For logs, one of LogNames, Attributes, Resources or Libraries must be specified with a
+	// non-empty value for a valid configuration.
+
+	// For metrics, one of MetricNames, Expressions, or ResourceAttributes must be specified with a
+	// non-empty value for a valid configuration.
+
+	// Services specify the list of items to match service name against.
+	// A match occurs if the span's service name matches at least one item in this list.
+	Services []string `river:"services,attr,optional"`
+
+	// SpanNames specify the list of items to match span name against.
+	// A match occurs if the span name matches at least one item in this list.
+	SpanNames []string `river:"span_names,attr,optional"`
+
+	// LogBodies is a list of strings that the LogRecord's body field must match against.
+	LogBodies []string `river:"log_bodies,attr,optional"`
+
+	// LogSeverityTexts is a list of strings that the LogRecord's severity text field must match against.
+	LogSeverityTexts []string `river:"log_severity_texts,attr,optional"`
+
+	// LogSeverityNumber defines how to match against a log record's SeverityNumber, if defined.
+	LogSeverityNumber *LogSeverityNumberMatchProperties `river:"log_severity_number,block,optional"`
+
+	// MetricNames is a list of strings to match metric name against.
+	// A match occurs if metric name matches at least one item in the list.
+	MetricNames []string `river:"metric_names,attr,optional"`
+
+	// Attributes specifies the list of attributes to match against.
+	// All of these attributes must match exactly for a match to occur.
+	// Only match_type=strict is allowed if "attributes" are specified.
+	Attributes []Attribute `river:"attribute,block,optional"`
+
+	// Resources specify the list of items to match the resources against.
+	// A match occurs if the data's resources match at least one item in this list.
+	Resources []Attribute `river:"resource,block,optional"`
+
+	// Libraries specify the list of items to match the implementation library against.
+	// A match occurs if the span's implementation library matches at least one item in this list.
+	Libraries []InstrumentationLibrary `river:"library,block,optional"`
+
+	// SpanKinds specify the list of items to match the span kind against.
+	// A match occurs if the span's span kind matches at least one item in this list.
+	SpanKinds []string `river:"span_kinds,attr,optional"`
+}
+
+// Convert converts args into the upstream type.
+func (args *MatchProperties) Convert() map[string]interface{} {
```

Comment:
ditto with unexporting the Convert methods here and elsewhere wherever they're defined